barcodescanner/src/main/java/com/mindera/skeletoid/barcodescanner/view/BarcodeScannerFragment.kt

```diff
@@ -0,0 +1,333 @@
+                            if (nextBarcodeDistanceToCenter < previousBarcodeDistanceToCenter) activeBarcode = next
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * Converts a barcode bounding box to a Rect object.
+     * a^2 + b^2 = c^2
+     *
+     */
+    private fun barcodeToRect(barcode: Barcode): Rect {
+        return Rect(barcode.boundingBox.left, barcode.boundingBox.top, barcode.boundingBox.left + barcode.boundingBox.right, barcode.boundingBox.top + barcode.boundingBox.bottom)
+    }
+
+    /**
+     * Distance to center using pythagoras theorem.
+     *
+     * We want the hypotenuse which is c^2, so, c = sqrt(a^2 + b^2)
+     *
+     */
+    private fun distanceToCenter(a: Rect, b: Rect) : Double {
+        return sqrt(Math.pow((a.centerX() - b.centerX()).toDouble(), 2.0) + Math.pow((a.centerY() - b.centerY()).toDouble(), 2.0))
+    }
+
+    /**
+     * Callback that is triggered when a new object is detected by the camera.
+     *
+     * @param data the barcode parsed object, in this case, our QRCode.
+     */
+    override fun onObjectDetected(data: Barcode?) {
+        //The callback can be triggered multiple times for the same object,
+        //So we might be looking at cases where, without synchronization,
+        //we are trying to find an item that does not yet exist in the list, but is already
+        //ready to be added, and our comparison will not return false, adding duplicates.
+        //To avoid that, RACE CONDITIONS!
+        synchronized(lock) {
+            //Avoid duplicates & null data..
+            if (data != null && readBarcodes.find { it.rawValue == data.rawValue } == null) {
+                //All good, add the new bar code.
+                readBarcodes.add(data)
+                //Get the closest barcode and save it on our active barcode live data.
+                if (::sourcePreviewMeasurements.isInitialized) getClosestBarcode(sourcePreviewMeasurements)
+                //Send message that we have a barcode so we can start the count down. ONLY one time.
+                if (hasBarcode.value == null) hasBarcode.postValue(true)
+            }
+        }
+    }
+
+    /**
+     * Receives the requestCameraPermission function result of success or lack of and handles it.
+     */
+    override fun onRequestPermissionsResult(requestCode: Int, permissions: Array<out String>, grantResults: IntArray) {
+        if (requestCode == RC_HANDLE_CAMERA_PERMISSION) {
+            if (grantResults.isNotEmpty() && grantResults.first() == PackageManager.PERMISSION_GRANTED) initCamera()
+            else Snackbar.make(snackbarView, R.string.barcode_scanner_fragment_no_camera_permission, Snackbar.LENGTH_SHORT).show()
+        } else {
+            super.onRequestPermissionsResult(requestCode, permissions, grantResults)
+        }
+    }
+
+    override fun onResume() {
+        Log.w(BARCODE_FRAGMENT_TAG, "Re-opening the camera onResume.")
```

Comment:
Please avoid using `Log`, always use `LOG` since it's Skeletoid default logger.

You already have Skeletoid base as a dependency, so it's just a simple change.

Reply:
done!